### ISSUE TITLE
Add or reassign partitions

### ISSUE THREAD

**Issue by oyld** [edenhill/librdkafka#461]
When, partitions increase or reassigned, consumer will know it?

**Comment by edenhill** [comment]
With the high-level KafkaConsumer you can register a rebalance_cb()   that will be called when the client receives an updated partition assignment from the group leader.

See here for an example:
https://github.com/edenhill/librdkafka/blob/master/examples/rdkafka_consumer_example.cpp#L125

And see docs here:
https://github.com/edenhill/librdkafka/blob/master/src-cpp/rdkafkacpp.h#L521

**Comment by oyld** [comment]
I have 2 consumers, each consume 3 partitions. When I add partition to 7, each consumer print a log:
2015-12-16 01:52:36.141: LOG-5-PARTCNT: Topic test_oyld partition count changed from 6 to 7
and then, they can't consume anything.

**Comment by edenhill** [comment]
Do all partitions have leaders after adding a partition?

**Comment by oyld** [comment]
Yes，the  replication factor is 1

**Comment by edenhill** [comment]
Okay, I'll try to reproduce

**Comment by edenhill** [comment]
Could you reproduce this with `debug=topic,broker,fetch` and upload the logs? Thanks

**Comment by oyld** [comment]
[Uploading out.txt…]()

**Comment by oyld** [comment]
[out.txt](https://github.com/edenhill/librdkafka/files/66403/out.txt)

**Comment by edenhill** [comment]
Are you on the latest master? (pushed a bunch of stuff yesterday)

**Comment by oyld** [comment]
Yes, I updated these today afternoon.

**Comment by gabriel-ibarra** [comment]
Hi, sorry for writing in a closed issue, but it is not clear to me if it is really an issue or not, or if it was fixed.

I'm using the librdkafka-0.9.1 and I'm seeing the same that oyld.

The steps that I follow are the next:

Having a topic with 1 partition, and a consumer for this topic.
I added a new partition dynamically while the consumer is running.
After that, I expected that a rebalance process were performed.
But, for a while, It only consume from the old partition, and after some seconds the next message is showed:
... "Topic TopicTest partition count changed from 1 to 2" ...
And finally after this message, no more message are received.

Thanks,
